### PR TITLE
train on multi-gpu, improve bbox plotting

### LINKED ISSUE #11: check transform tools
They seem to do crazy stuff that is not intended!

### PR DESCRIPTION
**Pull request by Bartzi**
MULTIGPU!!!
fixes #11

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -165,17 +178,30 @@ def main():
...
+    else:
+        updater = training.updaters.MultiprocessParallelUpdater(
+            train_iter, optimizer, devices=args.gpu)
+        updater.setup_workers()
+
     trainer = training.Trainer(updater, (120000, 'iteration'), args.out)
 
+    # trainer.extend(
+    #     extensions.ExponentialShift('lr', 0.1, init=1e-3),
+    #     trigger=triggers.ManualScheduleTrigger([80000, 100000], 'iteration'))
```
Comment (Jopyth) [inline train.py]: Is this something we might need in the future?

Comment (Bartzi) [reply train.py]: Hmm.... I'm not sure... maybe but we can also kill it

**Comment by Bartzi** [comment]
I added something else, do you want to look again?

**Comment by Bartzi** [comment]
fixes #11

File: datasets/sheep_dataset.py [hunk 2]
```diff
@@ -25,16 +25,19 @@ def __init__(self, dataset_root, label_file, image_size=(512, 512)):
...
         return len(self.data)
 
-    def get_image(self, i, do_resize=True):
-        image_path = os.path.join(self.dataset_root, self.data[i]['image'])
+    def load_image(self, image_path, resize_to=None):
         with Image.open(image_path) as image:
-            if do_resize:
-                image = image.resize(self.image_size)
+            if resize_to is not None:
+                image = image.resize(resize_to)
```
Comment (Jopyth) [inline datasets/sheep_dataset.py]: :+1:

File: train.py [hunk 3]
```diff
@@ -117,10 +123,13 @@ def main():
...
         '--model', choices=('ssd300', 'ssd512'), default='ssd512')
     parser.add_argument('--batchsize', type=int, default=32)
-    parser.add_argument('--gpu', type=int, default=-1)
+    parser.add_argument('--gpu', type=int, nargs='*', default=[])
     parser.add_argument('--out', default='result')
     parser.add_argument('--resume')
     parser.add_argument('--lr', type=float, default=0.001, help="default learning rate")
+    parser.add_argument('--port', type=int, default=1337, help="port for bbox sending")
+    parser.add_argument('--ip', default='127.0.0.1', help="destination ip for bbox sending")
+    parser.add_argument('--test-image', help="path to test image that shall be displayed in bbox vis")
```
Comment (Jopyth) [inline train.py]: :+1:

**Pull request merged**